lib/generate.m

```diff
@@ -0,0 +1,61 @@
+function [x] = generate(fs,t,q,friv,w,taus,a0,a)
+%generate is the final tool that helps us in generating sigma and delta
+%(longitudinal and transversal) signals related to the Schottky analysis
+
+n=size(t,1);
+dt=t(2)-t(1);
+intTime=t(end)+dt;
+nRect=round(intTime*friv);
+tau=((0:nRect)/friv)';
+
+x=zeros(n,1);
+
+if(fs~=0)
+    tau=tau+taus*sin(2*pi*fs*tau); %time modulation
+end
+
+if tau(end)/dt<n
+    add=1;
+else
+    add=0;
+end
+
+for k=1:nRect+add
+%     x=foo(tau(k),w,dt,t,x); %longitudinal bunched, maybe useful for gauss
+    ind=round(tau(k)/dt)+1;
+    x(ind)=1;
+end
+
+if(q~=0)
+    %modulation sin, transverse position
+    x=x.*(a0+a*sin(2*pi*q*friv*t)); %transversal bunched
+end
+
+end
+
+function [S] = foo(tau_1,w,dt,t,S)
```

Comment:
not exactly the most self-explaining name for a function. please change it to something more sensible